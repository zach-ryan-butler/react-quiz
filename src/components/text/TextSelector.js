import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ text, textHandler }) {
  return (
    <section>
      <input 
        value={text}
        placeholder="Full Name" 
        onChange={textHandler}/>
    </section>
  );
}

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  textHandler: PropTypes.func.isRequired
};

export default TextSelector;
