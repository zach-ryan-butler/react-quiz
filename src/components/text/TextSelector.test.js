import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('TextSelector component', () => {
  it('renders TextSelector', () => {
    const wrapper = shallow(<TextSelector
      text="Zach Butler"
      textHandler={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
