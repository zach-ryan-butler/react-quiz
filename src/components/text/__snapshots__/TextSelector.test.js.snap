// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextSelector component renders TextSelector 1`] = `
<section>
  <input
    onChange={[Function]}
    placeholder="Full Name"
    value="Zach Butler"
  />
</section>
`;
